pandas/core/generic.py

```diff
@@ -2576,18 +2576,27 @@ def to_sql(
         >>> engine.execute("SELECT * FROM users").fetchall()
         [(0, 'User 1'), (1, 'User 2'), (2, 'User 3')]
 
-        >>> df1 = pd.DataFrame({'name' : ['User 4', 'User 5']})
-        >>> df1.to_sql('users', con=engine, if_exists='append')
+        An `sqlalchemy.engine.Connection` can also be passed to to `con`:
+        >>> with engine.connect() as connection:
```

Comment:
`with engine.begin() as connection:` would be a safer bet. Upcoming changes in SQLAlchemy will (probably) result in both `engine.connect()` and `engine.begin()` returning a `Connection` with an active transaction, but the default behaviour on exit from the context manager block will (probably) be that for `.connect()` the transaction will be rolled back, while with `.begin()` the transaction will be committed.

Reply:
Sounds good - fixed it